### PR TITLE
Releasing improvements

### PR DESCRIPTION
**Pull request by qc00**
* #214
* #215
* Make LMDB less flaky on Windows
* `MsgPackNormalizer` avoid copying + thread-safe
* Some documentation on the Actions

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -6,9 +6,84 @@ on:
...
+
+            const cwd = process.cwd();
+            const client = new dhc.DownloadHttpClient();
+            const artifactsResponse = await client.listArtifacts();
+            for (const artifact of artifactsResponse.value) {
+                if (artifact.name.startsWith("build-metadata-follower-")) {
+                  console.log(`Downloading ${artifact.name} to ${cwd}`);
+                  const itemsResponse = await client.getContainerItems(artifact.name, artifact.fileContainerResourceUrl);
+                  // Must match "symbols" in build.yml
+                  const symbols_ext = artifact.name.search('windows') >= 0 ? ".pdb" : ".gz";
```
Comment (poodlewars) [inline .github/workflows/publish.yml:43]: Worth some assertions that the artifacts we expect are here?

**Review by poodlewars — APPROVED** [review]

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+<tr><th>vars.TWINE_USERNAME<br>or secrets.TWINE_USERNAME</th><td colspan="2">Please set API tokens, not real user names and passwords.</td>
+<tr><th>secrets.TWINE_PASSWORD</th>
+<tr><th>vars.TWINE_REPOSITORY</th><td>Well know repos like pypi OR testpypi</code>
+<tr><th>vars.TWINE_REPOSITORY_URL</th><td>Alternatively specify the URL of the repo</td>
+<tr><th>vars.TWINE_CERT</th><td>SSL CA</td>
+</table>
+
+## [docs.yml](docs.yml)
+
+**Runs on forks**: Yes. Must supply a CloudFlare Pages site to upload to
```
Comment (mehertz) [inline .github/workflows/README.md:48]: Have we tested this? Does this auto-upload to cloudflare?!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -29,7 +29,8 @@ jobs:
     steps:
       - run: |
           if ${{startsWith(github.ref, 'refs/tags/v')}}; then
-            echo -e "PUBLISH_ENV=ProdPypi\nCMAKE_PRESET_TYPE=release" | tee -a $GITHUB_ENV
+            # Can only upload to Pypi once per version, so only auto upload on tag builds:
+            echo -e "PUBLISH_ENV=ProdPypi\nCMAKE_PRESET_TYPE=release\nPYPI_PUBLISH=1" | tee -a $GITHUB_ENV
```
Comment (mehertz) [inline .github/workflows/build.yml:33]: We auto-upload? I thought we used the action?

File: .github/workflows/README.md (hunk 2 above)
Comment (qc00) [reply .github/workflows/README.md:48]: Of course. I have shown you a screenshot earlier

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    environment: ${{inputs.environment}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Detect system Python implementation
+        if: ${{!inputs.api_wheel}}
+        run: |
+          python3 -c 'import platform
+          prefix = {"CPython": "cp", "PyPy": "pp"}[platform.python_implementation()]
```
Comment (mehertz) [inline .github/workflows/docs.yml:22]: I don't think we are ever going to support PyPy...

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          node-version: 16
+      - run: npm install @actions/artifact
+        if: ${{!inputs.api_wheel}}
+      - id: download
+        name: Fetch wheel artifacts
+        if: ${{!inputs.api_wheel}}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const dhc = require("./node_modules/@actions/artifact/lib/internal/download-http-client.js");
```
Comment (mehertz) [inline .github/workflows/docs.yml:39]: Oh my word

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+      - name: Detect system Python implementation
+        if: ${{!inputs.api_wheel}}
+        run: |
+          python3 -c 'import platform
+          prefix = {"CPython": "cp", "PyPy": "pp"}[platform.python_implementation()]
+          v = platform.python_version_tuple()
+          print(f"PY_IMPL={prefix}{v[0]}{v[1]}")' | tee -a $GITHUB_ENV
+
+      # Copy and pasted from publsh.yml, until GitHub adds selective artifact download
```
Comment (mehertz) [inline .github/workflows/docs.yml:26]: Typo

File: .github/workflows/build_steps.yml [hunk 7]
```diff
@@ -230,14 +227,14 @@ jobs:
             MSYS_NO_PATHCONV=1 reg.exe ADD 'HKLM\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps'
```
[the remark this answers is not in the document]
Comment (qc00) [reply .github/workflows/build_steps.yml:227]: Oh, this was to figure out why we don't get crash dumps when the tests crash, but they no longer crash....

File: README.md [hunk 8]
```diff
@@ -37,7 +38,7 @@ ArcticDB is designed from the outset to be resilient; there is no single point o
 
 ## Quickstart
 
-> :warning: **Pre-built binaries only available for Linux**: Windows and MacOS binaries are coming soon!
+> :warning: **Pre-built binaries only available for Linux and Windows**: MacOS binaries are coming soon!
```
Comment (mehertz) [inline README.md:41]: 🥇

File: python/arcticdb/version_store/_normalization.py [hunk 9]
```diff
@@ -848,12 +851,13 @@ def normalize(self, obj, **kwargs):
...
 
-        d, r = divmod(self._buffer.tell(), 8)  # pack 8 by 8
+        d, r = divmod(buffer.tell(), 8)  # pack 8 by 8
         size = d + int(r != 0)
 
-        norm_meta.msg_pack_frame.size_bytes = self._buffer.tell()
+        norm_meta.msg_pack_frame.size_bytes = buffer.tell()
 
-        column_val = np.array(memoryview(self._buffer[: size * 8]), np.uint8).view(np.uint64)
+        # FUTURE(#263): do we need to care about byte ordering?
```
Comment (mehertz) [inline python/arcticdb/version_store/_normalization.py:859]: As in endianness?

File: python/tests/conftest.py [hunk 10]
```diff
@@ -203,6 +204,15 @@ def create_version_store(
 
     try:
         yield create_version_store
+    except RuntimeError as e:
+        if "mdb_" in str(e): # Dump keys
```
Comment (mehertz) [inline python/tests/conftest.py:208]: What's this?

File: .github/workflows/docs.yml (hunk 6 above)
Comment (qc00) [reply .github/workflows/docs.yml:26]: Errr, which one?

File: python/arcticdb/version_store/_normalization.py (hunk 9 above)
Comment (qc00) [reply python/arcticdb/version_store/_normalization.py:859]: Yes

File: python/tests/conftest.py (hunk 10 above)
Comment (qc00) [reply python/tests/conftest.py:208]: `mdb_` is the prefix that LMDB would have in its error messages. There's no specific exception I can catch here.

**Pull request merged**